packages/http-specs/specs/routes/mockapi.ts

```diff
@@ -1,11 +1,19 @@
-import { mockapi, passOnSuccess, ScenarioMockApi, ValidationError } from "@typespec/spec-api";
+import { MockRequest, passOnSuccess, ScenarioMockApi, ValidationError } from "@typespec/spec-api";
 
 export const Scenarios: Record<string, ScenarioMockApi> = {};
 
-function defineUri(uri: string) {
-  const url = new URL("http://example.com" + uri);
-  return passOnSuccess(
-    mockapi.get(url.pathname, (req) => {
+function createTests(uri: string, serverUri?: string, params?: Record<string, any>) {
+  return passOnSuccess({
+    uri: uri,
+    method: "get",
+    request: {
+      params: params,
```

Comment:
is this ncesssary if the uri already contains the params?

Reply:
@timotheeguerin  Yes. The call will fail if the request has the parameters and the values are not set.

Reply:
what fails? axios? `fetch` just works for that for example 
```
fetch("https://localhost:4000?a=3")
```

Reply:
Seems like an axios restriction

Reply:
@timotheeguerin I have refactored the code so the call to `createTests` does not have to pass in the parameters. It can be deduced with the uri. But, it still needs to be set in the parameters before making the call.

Let me explain. In the logic, we are using the same uri for both server and client testing. So, In order to register for client side execution (with `serve` command) we have to set the uri to pathname only. This is the logic followed even in cadl ranch. So, essentially the parameters in the uri is gone while using the same code for the server side execution (with `server-test` command). So, In order to accomodate for the lost parameters, we have to set the parameters.